test/unit_tests/arkode/C_serial/ark_test_sunadjoint.c

```diff
@@ -0,0 +1,408 @@
+/* -----------------------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2024, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------------------
+ * Program to test the SUNAdjoint capability with ARKODE. The test uses the
+ * Lotka-Volterra problem with four parameters as the test case.
+ * ---------------------------------------------------------------------------*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <sundials/sundials_core.h>
+
+#include <nvector/nvector_manyvector.h>
+#include <nvector/nvector_serial.h>
+#include <sunadjoint/sunadjoint_checkpointscheme_basic.h>
+#include <sunadjoint/sunadjoint_stepper.h>
+#include <sunmatrix/sunmatrix_dense.h>
+#include <sunmemory/sunmemory_system.h>
+
+#include <arkode/arkode.h>
+#include <arkode/arkode_arkstep.h>
+
+typedef struct
+{
+  sunrealtype tf;
+  sunrealtype dt;
+  int order;
+  int check_freq;
+  sunbooleantype save_stages;
+  sunbooleantype keep_checks;
+} ProgramArgs;
+
+static sunrealtype params[4] = {1.5, 1.0, 3.0, 1.0};
```

Comment:
Do these need `SUN_RCONST`?

Reply:
Since this in the `unit_tests` directory, yes, we should use them even though it should not matter with these values.